app/src/main/java/io/gnosis/safe/ui/transaction/TransactionViewHolderFactory.kt

```diff
@@ -38,64 +41,81 @@ class TransactionViewHolderFactory : BaseFactory<BaseTransactionViewHolder<Trans
-            is Transaction.ChangeMastercopyQueued -> TransactionViewType.CHANGE_MASTERCOPY_QUEUED
-            is Transaction.SettingsChangeQueued -> TransactionViewType.SETTINGS_CHANGE_QUEUED
-            is Transaction.TransferQueued -> TransactionViewType.TRANSFER_QUEUED
+            is TransactionView.ChangeMastercopy -> TransactionViewType.CHANGE_MASTERCOPY
+            is TransactionView.SettingsChange -> TransactionViewType.SETTINGS_CHANGE
+            is TransactionView.Transfer -> TransactionViewType.TRANSFER
+            is TransactionView.ChangeMastercopyQueued -> TransactionViewType.CHANGE_MASTERCOPY_QUEUED
+            is TransactionView.SettingsChangeQueued -> TransactionViewType.SETTINGS_CHANGE_QUEUED
+            is TransactionView.TransferQueued -> TransactionViewType.TRANSFER_QUEUED
             else -> throw UnsupportedViewType(javaClass.name)
         }.ordinal
 }
 
-abstract class BaseTransactionViewHolder<T : Transaction>(viewBinding: ViewBinding) : Adapter.ViewHolder<T>(viewBinding.root)
+
+abstract class BaseTransactionViewHolder<T : TransactionView>(viewBinding: ViewBinding) : Adapter.ViewHolder<T>(viewBinding.root)
 
 class ChangeMastercopyViewHolder(viewBinding: ItemTxChangeMastercopyBinding) :
-    BaseTransactionViewHolder<Transaction.ChangeMastercopy>(viewBinding) {
+    BaseTransactionViewHolder<TransactionView.ChangeMastercopy>(viewBinding) {
 
-    override fun bind(data: Transaction.ChangeMastercopy, payloads: List<Any>) {
+    override fun bind(data: TransactionView.ChangeMastercopy, payloads: List<Any>) {
         TODO("Not yet implemented")
     }
 }
 
-class SettingsChangeViewHolder(viewBinding: ItemTxSettingsChangeBinding) :
-    BaseTransactionViewHolder<Transaction.SettingsChange>(viewBinding) {
+class SettingsChangeViewHolder(private val viewBinding: ItemTxSettingsChangeBinding) :
+    BaseTransactionViewHolder<TransactionView.SettingsChange>(viewBinding) {
 
-    override fun bind(data: Transaction.SettingsChange, payloads: List<Any>) {
-        TODO("Not yet implemented")
+    override fun bind(data: TransactionView.SettingsChange, payloads: List<Any>) {
+        with(viewBinding) {
+            settingName.text = data.transaction.dataDecoded.method
+            dateTime.text = data.transaction.date
+        }
     }
 }
 
-class TransferViewHolder(viewBinding: ItemTxTransferBinding) :
-    BaseTransactionViewHolder<Transaction.Transfer>(viewBinding) {
-
-    override fun bind(data: Transaction.Transfer, payloads: List<Any>) {
-        TODO("Not yet implemented")
+class TransferViewHolder(private val viewBinding: ItemTxTransferBinding) :
+    BaseTransactionViewHolder<TransactionView.Transfer>(viewBinding) {
+
+    override fun bind(data: TransactionView.Transfer, payloads: List<Any>) {
+        with(viewBinding) {
+            val value = data.transaction.value
+            amount.text = "${value.shiftedString(18)} ${data.transaction.tokenInfo?.symbol}"
```

Comment:
we should extract this 18 to some constant with descriptive name

Reply:
We probably want a more comprehensive formatter here anyway. Should I mark this with a TODO? To fix this when we have the tokeninfo needed to decide how far to shift? Or is that the same for every token and Ether?

Reply:
So I can use `data.transaction.tokenInfo.decimals` instead of 18 here?

Reply:
This should come somehow from your lower layer of mapping. In your `TransactionView.Transfer`. I would also say that this is a bit outside scope as this should be addressed here: https://github.com/gnosis/safe-android/issues/663

Reply:
Just let Liliya know that the UI she might see out of transactions right now is not valid and shouldn't be tested until #663 is implemented.